sycl/doc/extensions/supported/sycl_ext_intel_buffer_location.asciidoc

```diff
@@ -86,6 +86,36 @@ It also notifies the SYCL runtime to store the given accessor in that memory. |
 |===
 --
 
+== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification section 6.3.3 "Feature test macros". Therefore, an implementation
+supporting this extension must predefine the macro `SYCL_EXT_INTEL_BUFFER_LOCATION` to
+one of the values defined in the table below. Applications can test for the
+existence of this macro to determine if the implementation supports this
+feature, or applications can test the macro’s value to determine which of the
+extension’s APIs the implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value |Description
+|1     |Initial extension version. Base features are supported.
+|===
+
+== Extension to `enum class aspect`
+
+[source]
+----
+namespace sycl {
+enum class aspect {
+  ext_intel_buffer_location
+}
```

Comment:
Rather than adding an aspect, it would be more convenient if the implementation simply ignored the property in cases where it makes no sense for the device.  This would be similar to what we did for the `init_mode` and `implement_in_csr` properties in [sycl_ext_oneapi_device_global](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_device_global.asciidoc).

I think we should look for ways to avoid adding lots of new aspects.  Otherwise, it will be very painful for applications because they will need to have tons of apsect-checking tests everywhere.

Reply:
> Should it be ignored by DPCPP RT or by plugin or by backend?

I don't think it matters from a user perspective.

> Would it be correct to change value of the property implicitly?

I don't understand this.  Are you asking if it is correct to ignore the property in the case when the device does not support it?

> User may also want to use different code in cases when device supports buffer_location feature and when it doesn't.

Rather than having tests for each feature like this, it seems like it would be better to say that features like `buffer_location`, `init_mode`, and `implement_in_csr` are supported on all FPGA devices, and we already have a way for a user to test whether a device is FPGA.  In addition, it seems reasonable to ignore these properties when used on a non-FPGA device, since their semantics have no meaning on other devices.  This makes it easier for users to pass the same kernel to any device when using these properties.

Reply:
> I don't understand this. Are you asking if it is correct to ignore the property in the case when the device does not support it?
I meant that if user passed some value to buffer_location property and this property is not supported by device, we will implicitly change this value to default and user will not know about it. Is it correct?

Reply:
Yes, I think I agree, but this seems like a strange way of saying it.  It would be clearer to say that the `bufffer_location` property is silently ignored if the device does not support that property.

Reply:
I added ignoring of the property in the main PR https://github.com/intel/llvm/pull/5604

Reply:
Great.  Can you also update the extension specification to state that the `buffer_location` property is ignored for non-FPGA devices?  You can use similar wording as what was added in #5665.